### DISCUSSION TITLE
Estampille qui contient un champ du formulaire

### DISCUSSION THREAD

**Discussion by ChrisThe01** [MTESSDev/FRW#264] [Annonces]
Bonjour,

Est-ce qu'il est possible de mettre dans l'estampille la valeur d'un champ depuis le formulaire FORM ? Si oui, pouvez-vous me fournir la syntaxe ?

Par exemple, j'ai un champ (caché) qui contient le numéro de la demande que je voudrais inscrire dans l'estampille pour le PDF produit.

Merci,

**Comment by loima02** [comment]
Bonjour,

Est-ce que l'exemple décrit dans la documentation du fichier bind répond à votre besoin?

https://github.com/MTESSDev/FRW/blob/main/Documentation/fichiers-bind.md#2--exemple-de-configuration-avec-un-gabarit-word-sp%C3%A9cifique

Merci et bonne journée!

**Comment by ChrisThe01** [comment]
Oui mais comment je peux référencier un champ du formulaire ? Quelle est la syntaxe ?

Christian Thériault
Analyste en informatique
Division versement et systèmes de soutien
Service des systèmes partagés et compte-client
Direction des systèmes partagés, du travail et de l'assurance parentale
Direction générale des technologies de l'Information
Ministère du Travail, de l'Emploi et de la Solidarité sociale

425, Jacques-Parizeau, 3e étage
Québec (Québec) G1R 4Z1
Téléphone (local) : 581-628-8934 poste 80869
Téléphone (sans frais) : 1-888-628-8934 poste 80869
***@***.******@***.***>
www.mtess.gouv.qc.ca<http://www.mtess.gouv.qc.ca/>
[Je me protège et je sauve des vies]
Ce courriel peut contenir des renseignements confidentiels en vertu de la Loi sur l'accès aux documents des organismes publics et sur la protection des renseignements personnels. Ces renseignements ne peuvent être utilisés que par le destinataire mentionné dans l'en-tête. Si ce courriel vous est parvenu par erreur, nous vous informons que son usage est formellement interdit. Vous devez signaler cette erreur dans les meilleurs délais en communiquant avec l'expéditeur et détruire ce courriel. Nous vous remercions de votre collaboration.

De : Mathieu Loiselle ***@***.***>
Envoyé : 5 juin 2024 15:20
À : MTESSDev/FRW ***@***.***>
Cc : Thériault, Christian ***@***.***>; Author ***@***.***>
Objet : [EXTERNE] Re: [MTESSDev/FRW] Estampille qui contient un champ du formulaire (Discussion #264)

*ATTENTION : Ce courriel provient de l'extérieur de votre organisation.
Évitez de cliquer sur un hyperlien, d'ouvrir une pièce jointe ou de transmettre des informations personnelles si vous ne connaissez pas l'expéditeur du courriel. En cas de doute, communiquez verbalement avec lui.

Bonjour,

Est-ce que l'exemple décrit dans la documentation du fichier bind répond à votre besoin?

https://github.com/MTESSDev/FRW/blob/main/Documentation/fichiers-bind.md#2--exemple-de-configuration-avec-un-gabarit-word-sp%C3%A9cifique

Merci et bonne journée!

-
Reply to this email directly, view it on GitHub<https://github.com/MTESSDev/FRW/discussions/264#discussioncomment-9681720>, or unsubscribe<https://github.com/notifications/unsubscribe-auth/A62QKPJ4LBTYR3DD2IWFDQLZF5QHPAVCNFSM6AAAAABI3KKEC6VHI2DSMVQWIX3LMV43SRDJONRXK43TNFXW4Q3PNVWWK3TUHM4TMOBRG4ZDA>.
You are receiving this because you authored the thread.Message ID: ***@***.******@***.***>>

________________________________
Ce message est confidentiel et est à l'usage exclusif du destinataire nommé ci-dessus. Si vous recevez ce message par erreur, sachez qu'il est interdit de le diffuser, de le distribuer, d'en dévoiler le contenu ou de le reproduire. Veuillez informer immédiatement l'expéditeur par courrier électronique et détruire l'original de ce message ainsi que toute copie.
________________________________

**Comment by loima02** [comment]
Bonjour,

Dans votre fichier transmissions (soit le fichier du formulaire, soit le fichier global à votre système autorisé), vous devriez avoir une ligne qui identifie le contenu de vos champsFormulaire. Dans nos exemples, on utilise `DonneesFormulaire`

```
      content:
        json_content: |
          {
            "entree": {
                "champsFormulaire": {{{Json DonneesFormulaire ">flatten;_;_{0}" ">forceString"}}},
```

Lorsque vous complétez le formulaire, à l'aide des outils de développement, ouvrez le `Contenu du modèle de données` pour trouver le champ que vous souhaitez afficher

```
{
  "form": {
    "validAll": false,
    "EtatRevision": "sans-erreur",
    "PjManquantes": [],
    "NomFamille": "test",
```

Dans la section `estampille` de votre fichier bind, vous pouvez alors référer à votre champ de formulaire. Par exemple `DonneesFormulaire`.`form`.`NomFamille`

```
    estampille:
      positionX: 400
      positionY: 710
      tailleFont: 9
      lignes:
        - '{{{DonneesFormulaire.form.NomFamille}}}'
```

L'estampille affiche maintenant le nom de famille inscrit par l'utilisateur.

**Comment by ChrisThe01** [comment]
Merci ! C'est exactement ce que je cherchais.

**Reply by loima02** [reply]
Bonjour,

J'ai ajouté un exemple dans la documentation du fichier bind (réf. : https://github.com/MTESSDev/FRW/blob/main/Documentation/fichiers-bind.md#2--exemple-de-configuration-avec-un-gabarit-word-sp%C3%A9cifique).

Si cette solution vous convient, vous pouvez identifier ma réponse ci-dessus comme "Réponse" à la discussion en cours et fermer la discussion.

Merci et bonne journée.

**Reply by ChrisThe01** [reply]
Réponse : C'est parfait.
